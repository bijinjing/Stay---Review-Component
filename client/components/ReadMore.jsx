import React from 'react';


class ReadMore extends React.Component{
  constructor(props) {
    super(props)

    this.state={
      expand:false
    }

    this.expandHandler = this.expandHandler.bind(this)
  }

  expandHandler(){
    let oldState = this.state.expand;
    let expand = oldState? false: true
    this.setState({
      expand
    })
  }

  render(){
    return(
       this.state.expand? <div className="read_more">{this.props.comments}</div>:
        <div className="read_more">
          <a>{this.props.comments.split(" ").slice(0,30).join(" ") + "..."}</a><a onClick={this.expandHandler}>Read more</a>
        </div>
    )
  }
}

export default ReadMore