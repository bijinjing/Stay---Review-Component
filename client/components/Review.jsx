import React from 'react'

class Review extends React.Component{
  constructor(props) {
    super(props)

  }

  render(){
    return(
      <div>
        <img src={this.props.data.profile}></img>
        <div>{this.props.data.name}</div>
        <div>{this.props.data.review_date}</div>
        <div>{this.props.data.comments}</div>
      </div>
    )
  }
}

export default Review