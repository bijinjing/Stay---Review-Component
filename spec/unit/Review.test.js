import React from 'react';
import Review from '../../client/components/Review.jsx';
import ReadMore from '../../client/components/ReadMore.jsx';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme'
Enzyme.configure({ adapter: new Adapter() });

describe('Review', ()=>{
  it('renders without crashing', () => {
    const props = {
      data:{
        profile:'https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/128.jpg',
        name:"jinjing",
        review_data:"3/14/19",
        comments:"it is good"
      },
      searchStatus:false
    }

    const wrapper = shallow(<Review {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})



describe('Review_ReadMore', () => {
  it('Render ReadMore component when comments has over 30 words',() => {
    const props={
      data:{
        profile:'https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/128.jpg',
        name:"jinjing",
        review_data:"3/14/19",
        comments:"Necessitatibus dolorem eius nemo odit aut quisquam natus. Quam ullam vitae fugit amet. Aut sequi repudiandae voluptas. Ducimus velit sequi iusto nemo ad accusantium et molestias. Atque quibusdam iste doloremque ea. In sint sed voluptatem unde fghj rdftyg drtf ghj ftyg hjk rytu"
      },
      searchStatus:false
    }

    const wrapper = shallow(<Review {...props}/>);
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})






