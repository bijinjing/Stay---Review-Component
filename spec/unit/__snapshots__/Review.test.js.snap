// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review renders without crashing 1`] = `
<div>
  <img
    src="https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/128.jpg"
  />
  <div>
    jinjing
  </div>
  <div />
  <div>
    it is good
  </div>
</div>
`;

exports[`Review_ReadMore Render ReadMore component when comments has over 30 words 1`] = `
<div>
  <img
    src="https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/128.jpg"
  />
  <div>
    jinjing
  </div>
  <div />
  <ReadMore
    comments="Necessitatibus dolorem eius nemo odit aut quisquam natus. Quam ullam vitae fugit amet. Aut sequi repudiandae voluptas. Ducimus velit sequi iusto nemo ad accusantium et molestias. Atque quibusdam iste doloremque ea. In sint sed voluptatem unde fghj rdftyg drtf ghj ftyg hjk rytu"
  />
</div>
`;
